modules/tinymce/src/core/main/ts/mode/Disabled.ts

```diff
@@ -0,0 +1,144 @@
+import { Arr, Optional, Strings } from '@ephox/katamari';
+import { Attribute, Compare, SelectorFilter, SelectorFind, SugarElement } from '@ephox/sugar';
+
+import Editor from '../api/Editor';
+import VK from '../api/util/VK';
+import * as ModeUtils from '../util/ModeUtils';
+
+const isDisabled = (editor: Editor): boolean => editor.disabled;
+
+const internalContentEditableAttr = 'data-mce-contenteditable';
+
+const setContentEditable = (elm: SugarElement<HTMLElement>, state: boolean) => {
+  elm.dom.contentEditable = state ? 'true' : 'false';
+};
+
+const switchOffContentEditableTrue = (elm: SugarElement<Node>) => {
+  Arr.each(SelectorFilter.descendants<HTMLElement>(elm, '*[contenteditable="true"]'), (elm) => {
+    Attribute.set(elm, internalContentEditableAttr, 'true');
+    setContentEditable(elm, false);
+  });
+};
+
+const switchOnContentEditableTrue = (elm: SugarElement<Node>) => {
+  Arr.each(SelectorFilter.descendants<HTMLElement>(elm, `*[${internalContentEditableAttr}="true"]`), (elm) => {
+    Attribute.remove(elm, internalContentEditableAttr);
+    setContentEditable(elm, true);
+  });
+};
+
+const toggleDisabled = (editor: Editor, state: boolean): void => {
+  const body = SugarElement.fromDom(editor.getBody());
+
+  if (state) {
```

Comment:
I know this will work but is it against convention to rely on truthiness? I thought we had to be explicit (e.g. `state === true`)? 

Just a nitpick in any case 😅

Reply:
I don't think you need to do that for booleans since you are not hiding weird behaviour it's either true or false a truthy check on other types is more vague what counts as true/false and then I think it's better to use type checks or similar o be very specific on what you are checking for.

Reply:
@spocke Thanks for clarifying!